assert_test.go

```diff
@@ -0,0 +1,83 @@
+package taskflow_test
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func failedMessage(message ...string) string {
+	if len(message) == 0 {
+		return "Failed"
+	}
+	return strings.Join(message, ",")
+}
+
+func assertTrue(t testing.TB, value bool, message ...string) {
+	if !value {
+		t.Helper()
+		t.Error(failedMessage(message...))
+	}
+}
+
+func assertContains(t testing.TB, value string, expected string, message ...string) {
+	if !strings.Contains(value, expected) {
+		t.Helper()
+		t.Errorf("%s\nHave [%s], want [%s]", failedMessage(message...), value, expected)
+	}
+}
+
+func requireEqual(t testing.TB, expected interface{}, value interface{}, message ...string) {
+	if !reflect.DeepEqual(value, expected) {
+		t.Helper()
+		t.Fatalf("%s\nHave [%v], want [%v]", failedMessage(message...), value, expected)
+	}
+}
+
+func assertEqual(t testing.TB, expected interface{}, value interface{}, message ...string) {
+	if !reflect.DeepEqual(value, expected) {
+		t.Helper()
+		t.Errorf("%s\nHave [%v], want [%v]", failedMessage(message...), value, expected)
```

Comment:
Very often I see the `got` and `want` (examples: [1](https://golang.org/pkg/testing/), [2](https://gobyexample.com/testing), [3](https://github.com/golang/go/blob/master/src/flag/flag_test.go#L477))

```
t.Errorf("%s\ngot: %v\nwant: %v", msg, value, want)
```

The same comment applies to other functions as well.